src/main/java/com/github/luben/zstd/ZstdInputStream.java

```diff
@@ -46,10 +46,24 @@
 
     // The main constructor / legacy version dispatcher
     public ZstdInputStream(InputStream inStream) {
+        this(inStream, BufferPool.get(srcBuffSize), null);
+    }
+
+    /**
+     * construct a ZstdInputStream with custom byte array. This construction is useful when you prefer to be able to
+     * manage the buffers. You can call {@link ZstdInputStream#srcBuffSize} to get the recommended size.
+     * @param inStream input stream
+     * @param src byte array
+     */
+    public ZstdInputStream(InputStream inStream, byte[] src) {
+        this(inStream, null, src);
+    }
+
+    private ZstdInputStream(InputStream inStream, BufferPool pool, byte[] src) {
```

Comment:
Why do we need this constructor?  I guess if buffer is supplied we are not going to need the pool.

Reply:
I try to avoid duplicate code in constructor. This construction is private and other public constructions depend on it. Or you prefer to remove this private constructor?

Reply:
I was imagining that the `BufferPool` constructor will just checkout a buffer and reuse the `byte[]` constructor. BTW, I didn't notice it was private, so I think it still acceptable.

Reply:
> I was imagining that the BufferPool constructor will just checkout a buffer and reuse the byte[] constructor. 

We have to keep a reference of BufferPool in order to checkIn the byte[] in closing stream. Hence, BufferPool constructor can reuse byte[] constructor.

Reply:
Yes, you are right. Let's keep it then as you proposed.